crates/build/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+use anyhow::{Context, Result};
+use path_absolutize::Absolutize;
+use spin_loader::local::config::{RawAppManifest, RawComponentManifest};
+use std::path::Path;
+use subprocess::Exec;
+use tracing::log;
+
+/// If present, run the build command of each component.
+pub async fn build(app: RawAppManifest, src: &Path) -> Result<()> {
+    let src = src.absolutize()?;
+    futures::future::join_all(
+        app.components
+            .into_iter()
+            .map(|c| build_component(c, &src))
+            .collect::<Vec<_>>(),
+    )
+    .await;
+
+    Ok(())
+}
+
+/// Run the build command of the component.
+async fn build_component(raw: RawComponentManifest, src: impl AsRef<Path>) -> Result<()> {
+    match raw.build {
+        Some(b) => {
+            log::info!("Executing the build command for component {}.", raw.id);
+            let first = b
+                .command
+                .first()
+                .context("Component build command must have at least one item.")?;
+
+            Exec::cmd(first)
+                .cwd(
```

Comment:
Doesn't need to be this PR, but it would be nice to be able to override this with e.g. `component.build.workdir`

Reply:
Opened #406 to track this.